.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,44 @@
+name: Build and Push Docker image to GHCR
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository }}
+          tags: type=raw,value=latest,enable={{is_default_branch}}
```

Comment:
:thought_balloon: I'm not familiar with this action but according to the [documentation](https://github.com/docker/metadata-action) it looks like this'll always update the tag as `latest` upon each push which makes sense but the documentation states that it supports semantic versioning to via the `semver` metadata. I think that means this should include `semver` in order to produce the equivalent tag for the Docker image when a new tag is created for this repository?

Reply:
This is actually building for quite some time because I included ARM64. I can remove that for now and the build time should be fairly short (2min vs 16 min on my fork). I added the semver tag as well. That typo was me trying to exit vim 😅 

Let me know if you still want any changes!

Reply:
Yeah, I think I've seen the ARM 64 builds take a long time before on a different project and I didn't ever get time to investigate why. If I build for multiple CPUs locally using OrbStack it's not nearly as long. Ah, Vim, funny!

OK, I'll fold in these changes. Thanks!